pkg/chains/storage/oci/legacy.go

```diff
@@ -280,8 +281,11 @@ func (b *Backend) RetrieveArtifact(ctx context.Context, obj objects.TektonObject
 
 func newDigest(cfg config.Config, imageName string) (name.Digest, error) {
 	// Override image name from config if set.
-	if r := cfg.Storage.OCI.Repository; r != "" {
-		imageName = r
+	if storageOCIRepository := cfg.Storage.OCI.Repository; storageOCIRepository != "" {
+		parts := strings.SplitN(imageName, "/", 3)
```

Comment:
I'm not sure if this covers the different possible use cases. `registry.com/spam/spam/spam/spam/spam/spam` is a perfectly valid repo name in most registries.

Reply:
Yes I agree with this, probably I should find a better way to make it more generalize

Reply:
Yes I agree with this, probably I should find a better way to make it more generalize

Reply:
I'm not sure what you mean. In the example above, `registry.com` is the registry, and `spam/spam/spam/spam/spam/spam` is the repo name. Some registries may treat the first `spam` as a username, but that's not globally true.

Reply:
Yes, you are right, there was a gap in my understanding, I was not aware that user is part of repo
I have updated the code accordingly